ymca/src/main/java/servlets/WebcamServlet.java

```diff
@@ -77,27 +79,36 @@ else if (null != data)
 			String decodedData = java.net.URLDecoder.decode(data, "UTF-8");
 			decodedData = decodedData.replace(' ', '+');
 			final byte[] imagedata = DatatypeConverter
-			        .parseBase64Binary(decodedData.substring(decodedData.indexOf(",") + 1));
+					.parseBase64Binary(decodedData.substring(decodedData.indexOf(",") + 1));
 
 			// send image to listeners
 			this.notifyObservers(imagedata);
 
+			LoggingEndpoint.log(id, Level.FINE, "Sending image to Vision Service");
```

Comment:
sessionId should be retrieved from the request object `request.getSession().getId();`